File: src/main/java/kr/hyfata/zero/ZeroConfig.java
package kr.hyfata.zero;

import kr.hyfata.zero.config.IConfig;
import kr.hyfata.zero.config.MailboxConfig;
import kr.hyfata.zero.config.ScoreboardConfig;
import org.bukkit.plugin.java.JavaPlugin;

public class ZeroConfig {
    IConfig scoreboardConfig = new ScoreboardConfig();
    IConfig mailboxConfig = new MailboxConfig();

    public ZeroConfig(JavaPlugin plugin) {
        scoreboardConfig.init(plugin);
        mailboxConfig.init(plugin);
    }

    public void save() {
        scoreboardConfig.saveConfig();
        mailboxConfig.saveConfig();
    }

    public IConfig getScoreboardConfig() {
        return scoreboardConfig;
    }

    public IConfig getMailboxConfig() {
        return mailboxConfig;
    }
}

File: src/main/java/kr/hyfata/zero/ZeroConfig.kt
package kr.hyfata.zero

import kr.hyfata.zero.config.AbstractConfig
import kr.hyfata.zero.modules.mailbox.MailboxConfig
import kr.hyfata.zero.modules.scoreboard.ScoreboardConfig
import org.bukkit.plugin.java.JavaPlugin

class ZeroConfig(plugin: JavaPlugin) {
    var scoreboardConfig: AbstractConfig = ScoreboardConfig()
    var mailboxConfig: AbstractConfig = MailboxConfig()

    init {
        scoreboardConfig.init(plugin)
        mailboxConfig.init(plugin)
    }

    fun save() {
        scoreboardConfig.saveConfig()
        mailboxConfig.saveConfig()
    }
}
